addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1387,6 +1427,26 @@ export class ListRenderer extends Component {
                     toFocus = buttons[index - 1] || currentButton;
                 }
                 break;
+            case "shift+arrowdown": {
+                this.expandCheckboxes(record, "down");
+                toFocus = this.findFocusFutureCell(cell, cellIsInGroupRow, "down");
+                break;
+            }
+            case "shift+arrowup": {
+                this.expandCheckboxes(record, "up");
+                toFocus = this.findFocusFutureCell(cell, cellIsInGroupRow, "up");
+                break;
+            }
+            case "shift+space":
+                this.toggleRecordSelection(record);
+                toFocus = cell;
+                break;
+            case "shift":
+                this.firstArrowShiftRecord = {
+                    record,
+                    index: this.props.list.records.indexOf(record),
+                };
```

Comment:
I find the name `shiftKeyedRecord` more appropriate for these reasons:
- not semantically bound to arrows
- I think it is misleading to call it "first" or "last" or something as it is just "the record on which we pressed the shift key and only that key"
- it it shorter :-)

Also, I don't like the object here. The name suggest this is a record but this is not. Why not compute the index where it is needed and simply store the record here ?
```suggestion
this.shiftKeyedRecord = record;
```

Reply:
Agree on the name 👍 
The reason I store the index of the record is that I don't want it to be computed every time the user expands its selection with shift+arrow. Depending on `limit` imposed in the arch or the pager limit of the list view it can be long every time the user expands